### PR TITLE
Use fqcr from command module invocation using shell module

### LINKED ISSUE #69788: Module redirection fails within collection for command if shell module is used in role within collection
##### SUMMARY

The test case https://github.com/ansible-collections/cisco.ios/blob/master/tests/integration/targets/ios_user/tests/cli/auth.yaml#L63 with task
```
    - name: test sshkey login
      shell: ssh ssh_user@{{ ansible_ssh_host }} -p {{ ansible_ssh_port|default(22)
        }} -o IdentityFile={{ role_path }}/files/test_rsa -o UserKnownHostsFile=/dev/null
        -o StrictHostKeyChecking=no -o BatchMode=yes -o PubkeyAuthentication=yes
        show version

```
fails with error
```
}
2020-05-22 18:07:38,267 p=27402 u=zuul n=ansible | TASK [ios_user : test sshkey login] ********************************************
2020-05-22 18:07:38,267 p=27402 u=zuul n=ansible | task path: /home/zuul/.ansible/collections/ansible_collections/cisco/ios/tests/integration/targets/ios_user/tests/cli/auth.yaml:63
2020-05-22 18:07:38,398 p=27974 u=zuul n=ansible | <162.253.42.96> attempting to start connection
2020-05-22 18:07:38,399 p=27974 u=zuul n=ansible | <162.253.42.96> using connection plugin ansible.netcommon.network_cli
2020-05-22 18:07:38,399 p=27974 u=zuul n=ansible | Found ansible-connection at path /home/zuul/venv/bin/ansible-connection
2020-05-22 18:07:40,807 p=27974 u=zuul n=ansible | <162.253.42.96> found existing local domain socket, using it!
2020-05-22 18:07:40,808 p=27974 u=zuul n=ansible | <162.253.42.96> ssh connection done, setting terminal
2020-05-22 18:07:40,808 p=27974 u=zuul n=ansible | <162.253.42.96> loaded terminal plugin for network_os cisco.ios.ios
2020-05-22 18:07:40,844 p=27974 u=zuul n=ansible | <162.253.42.96> updating play_context for connection
2020-05-22 18:07:40,844 p=27974 u=zuul n=ansible | <162.253.42.96> 
2020-05-22 18:07:40,844 p=27974 u=zuul n=ansible | <162.253.42.96> local domain socket path is /home/zuul/.ansible/pc/1e21ebcf11
2020-05-22 18:07:40,848 p=27974 u=zuul n=ansible | <162.253.42.96> ESTABLISH LOCAL CONNECTION FOR USER: zuul
2020-05-22 18:07:40,849 p=27974 u=zuul n=ansible | <162.253.42.96> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/zuul/.ansible/tmp/ansible-local-27402rj7dtu3e `"&& mkdir /home/zuul/.ansible/tmp/ansible-local-27402rj7dtu3e/ansible-tmp-1590170860.8482132-27974-108949243179715 && echo ansible-tmp-1590170860.8482132-27974-108949243179715="` echo /home/zuul/.ansible/tmp/ansible-local-27402rj7dtu3e/ansible-tmp-1590170860.8482132-27974-108949243179715 `" ) && sleep 0'
2020-05-22 18:07:40,857 p=27974 u=zuul n=ansible | redirecting (type: modules) cisco.ios.command to cisco.ios.ios_command
2020-05-22 18:07:41,138 p=27974 u=zuul n=ansible | Using module file /home/zuul/.ansible/collections/ansible_collections/cisco/ios/plugins/modules/ios_command.py
2020-05-22 18:07:41,139 p=27974 u=zuul n=ansible | <162.253.42.96> PUT /home/zuul/.ansible/tmp/ansible-local-27402rj7dtu3e/tmpbyt4_0pk TO /home/zuul/.ansible/tmp/ansible-local-27402rj7dtu3e/ansible-tmp-1590170860.8482132-27974-108949243179715/AnsiballZ_ios_command.py
2020-05-22 18:07:41,140 p=27974 u=zuul n=ansible | <162.253.42.96> EXEC /bin/sh -c 'chmod u+x /home/zuul/.ansible/tmp/ansible-local-27402rj7dtu3e/ansible-tmp-1590170860.8482132-27974-108949243179715/ /home/zuul/.ansible/tmp/ansible-local-27402rj7dtu3e/ansible-tmp-1590170860.8482132-27974-108949243179715/AnsiballZ_ios_command.py && sleep 0'
2020-05-22 18:07:41,146 p=27974 u=zuul n=ansible | <162.253.42.96> EXEC /bin/sh -c '/home/zuul/venv/bin/python /home/zuul/.ansible/tmp/ansible-local-27402rj7dtu3e/ansible-tmp-1590170860.8482132-27974-108949243179715/AnsiballZ_ios_command.py && sleep 0'
2020-05-22 18:07:41,409 p=27974 u=zuul n=ansible | <162.253.42.96> EXEC /bin/sh -c 'rm -f -r /home/zuul/.ansible/tmp/ansible-local-27402rj7dtu3e/ansible-tmp-1590170860.8482132-27974-108949243179715/ > /dev/null 2>&1 && sleep 0'
2020-05-22 18:07:41,419 p=27402 u=zuul n=ansible | fatal: [ios-15.6-2T]: FAILED! => {
    "changed": false,
    "invocation": {
        "module_args": {
            "_raw_params": "ssh ssh_user@162.253.42.96 -p 22 -o IdentityFile=/home/zuul/.ansible/collections/ansible_collections/cisco/ios/tests/integration/targets/ios_user/files/test_rsa -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -o BatchMode=yes -o PubkeyAuthentication=yes show version",
            "_uses_shell": true,
            "warn": true
        }
    },
    "msg": "Unsupported parameters for (command) module: _raw_params, _uses_shell, warn Supported parameters include: commands, interval, match, provider, retries, wait_for"
}
```
There is a short name redirection within the collection from `command` and `ios_command`
https://github.com/ansible-collections/cisco.ios/blob/c6aa603a701ab7d31b71e0b3dd4b08cffc738376/meta/runtime.yml#L134

```

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
<!--- Write the short name of the module, plugin, task or feature below, use your best guess if unsure -->

##### ANSIBLE VERSION
<!--- Paste verbatim output from "ansible --version" between quotes -->
```paste below
2.10
```

##### CONFIGURATION

```paste below

```

##### STEPS TO REPRODUCE

```yaml

```

##### ACTUAL RESULTS

```paste below

```

### PR DESCRIPTION
**Pull request by ganeshrn**
##### SUMMARY

Fixes https://github.com/ansible/ansible/issues/69788

Use fully qualified collection reference while invoking
command module from shell module

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

shell
##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by nitzmahone** [comment]
IMO this is the "right thing" anywhere we call a module/action internally, though we've sorta allowed people to replace the builtins in the past with local `library/` (etc) versions. I think the original issue is caused by something this doesn't address though: these internal redirections, even if we want to continue to allow non-collections overrides, should *never* be using the collections search path to find things. But, it looks like they are, since it found something outside the core routing metadata- I'll look into that.

**Review by bcoca — CHANGES_REQUESTED** [review]
As already mentioned, this breaks backwards compatibility.

**Comment by nitzmahone** [comment]
Hrm, looking at that closer, it's not even passing `collection_list` to `PluginLoader.get()`, so it shouldn't be possible for it to find that override... @ganeshrn are you sure you don't have an ansible-base routing entry for a command action/module, or a local library `command` module somewhere?

I'm not seeing how this could hit the behavior you're describing, and if it is, there's a nastier bug lurking...

For these internal cases, we might want to consider deprecating the local override capability (eg overriding `normal` on action- if you want to do that, use your own action).

**Comment by nitzmahone** [comment]
As I mentioned over on #69788 - I think the actual "right thing" to do here is to change this ref to `ansible.legacy.command`- that will do the legacy library search, followed by a fallback search to `ansible.builtin`, which is exactly the behavior you want here to preserve the legacy local override capability (and the reason `ansible.legacy` exists).

**Comment by bcoca** [comment]
this is still patching the bug for a single plugin while leaving it existing for the rest, while im not against merging this AFTER we fix the actual bug, it is not a full fix.

File: lib/ansible/plugins/action/shell.py [hunk 1]
```diff
@@ -17,7 +17,7 @@ def run(self, tmp=None, task_vars=None):
         self._task.action = 'command'
         self._task.args['_uses_shell'] = True
 
-        command_action = self._shared_loader_obj.action_loader.get('command',
+        command_action = self._shared_loader_obj.action_loader.get('ansible.builtin.command',
```
Comment (bcoca) [inline lib/ansible/plugins/action/shell.py]: s/ansible.builtin/ansible.legacy/

**Review by bcoca — CHANGES_REQUESTED** [review]

File: lib/ansible/plugins/action/shell.py (hunk 1 above)
Comment (ganeshrn) [reply lib/ansible/plugins/action/shell.py]: Fixed

**Pull request merged**